csharp/src/Vista.SDK/GmodNode.cs

```diff
@@ -36,6 +38,34 @@ internal GmodNode(GmodNodeDto dto)
 
     public GmodNode WithoutLocation() => Location is null ? this : this with { Location = null };
 
+    public GmodNode WithLocation(string location)
+    {
+        var locations = VIS.Instance.GetLocations(VisVersion);
+
+        return this with
+        {
+            Location = locations.CreateLocation(location)
+        };
+    }
+
+    public GmodNode TryWithLocation(string? location)
+    {
+        if (string.IsNullOrWhiteSpace(location))
+            return this;
+
+        var locations = VIS.Instance.GetLocations(VisVersion);
+        return this with { Location = locations.TryCreateLocation(location) };
```

Comment:
If `TryCreateLocation` fails, `Location` will be set to `null`, not sure if thats right